Preprocessing/oxygenPP.m

```diff
@@ -0,0 +1,214 @@
+
+% no modification of data is performed on the raw FV00 dataset except
+% local time to UTC conversion
+if strcmpi(qcLevel, 'raw'), return; end
+
+% auto logical in input to enable running under batch processing
+if nargin<3, auto=false; end
+
+for k = 1:length(sample_data)
+  
+  sam = sample_data{k};
+  
+  % Get the temperature
+  tempIdx       = getVar(sam.variables, 'TEMP');
+  if ~(tempIdx), continue; end % if there is no temperature, then don't do anythin  
+  % get coorinates and dimensions from TEMP to use as output
+  % coordinates/dimensions
+  dimensions = sam.variables{tempIdx}.dimensions;
+  if isfield(sam.variables{tempIdx}, 'coordinates')
+      coordinates = sam.variables{tempIdx}.coordinates;
+  else
+      coordinates = '';
+  end
+  temp = sam.variables{tempIdx}.data;
+  
+  % indexs of other variables that might be needed
+  psalIdx       = getVar(sam.variables, 'PSAL');
+  
+  [presRel, presName] = presRelFromSampleData(sam);
+
+	% from imosParameters.txt
+	% DOX,                 0, volume_concentration_of_dissolved_molecular_oxygen_in_sea_water,                          ml l-1,        ,              ,                                  O, 999999.0, 0.0,      200.0,    float
+	% DOX1,                1, mole_concentration_of_dissolved_molecular_oxygen_in_sea_water,                            umol l-1,      ,              ,                                  O, 999999.0, 0.0,      1000.0, float
+	% DOX2,                1, moles_of_oxygen_per_unit_mass_in_sea_water,                                               umol kg-1,     ,              ,                                  O, 999999.0, 0.0,      1000.0, float
+	% DOXS,                1, fractional_saturation_of_oxygen_in_sea_water,                                             percent,       ,              ,                                  O, 999999.0, ,         ,         float
+	% DOXY,                1, mass_concentration_of_oxygen_in_sea_water,                                                kg m-3,        ,              ,                                  O, 999999.0, 0.0,      29.0,     float
+	% DOXY_TEMP,           1, temperature_of_sensor_for_oxygen_in_sea_water,                                            degrees_Celsius,,             ,                                  T, 999999.0, 0.0,      50.0,     float
+
+  % oxygen parameters that maybe in the dataset
+  doxIdx        = getVar(sam.variables, 'DOX');
+  dox1Idx       = getVar(sam.variables, 'DOX1');
+  dox2Idx       = getVar(sam.variables, 'DOX2');
+  doxsIdx       = getVar(sam.variables, 'DOXS');
+  doxyIdx       = getVar(sam.variables, 'DOXY');
+  
+  % check is psal and some dox measurement present, if not skip
+  if ~(psalIdx && (doxIdx || dox1Idx || dox2Idx || doxsIdx)), continue; end
+    
+  psal = sam.variables{psalIdx}.data;
+  
+  ptmp = sw_ptmp(psal, temp , presRel, 0);
+  
+  % calculate the density at 0 dbar
+  dens = sw_dens0(psal, ptmp);
```

Comment:
I'm thinking about replacing calls to EOS-80 to equivalent in TEOS-10 in order to compute potential temperature and potential density. Something like:
```Matlab
  if ~isempty(sam.geospatial_lat_min) && ~isempty(sam.geospatial_lat_max) && ...
          ~isempty(sam.geospatial_lon_min) && ~isempty(sam.geospatial_lon_max)
      if sam.geospatial_lat_min == sam.geospatial_lat_max
          meanLat = sam.geospatial_lat_min;
      else
          meanLat = sam.geospatial_lat_min + ...
              (sam.geospatial_lat_max - sam.geospatial_lat_min)/2;
      end
      if sam.geospatial_lon_min == sam.geospatial_lon_max
          meanLon = sam.geospatial_lon_min;
      else
          meanLon = sam.geospatial_lon_min + ...
              (sam.geospatial_lon_max - sam.geospatial_lon_min)/2;
      end
  else
      disp(['Warning : no geospatial_lat_min/geospatial_lon_min documented for oxygenPP to be applied on ' sam.toolbox_input_file]);
      prompt = {'Latitude (South -ve)', 'Longitude (West -ve)'};
      dlg_title = 'Coords';
      num_lines = 1;
      defaultans = {'0', '0'};
      answer = inputdlg(prompt,dlg_title,num_lines,defaultans);
      
      % don't try to apply any correction if canceled by user
      if isempty(answer), continue; end
      
      meanLat = str2double(answer(1));
      meanLon = str2double(answer(2));
  end
  SA = gsw_SA_from_SP(psal, presRel, meanLon, meanLat);
  potTemp = gsw_pt0_from_t(SA, temp, presRel);
```
and
```Matlab
  CT = gsw_CT_from_pt(SA, potTemp);
  potDens = gsw_rho(SA, CT, 0);
```